packages/boot/tools/supports.ts

```diff
@@ -909,23 +909,89 @@ export const fetchCoreEvalRelease = async (
   planUrl = `${artifacts}/${config.name}-plan.json`,
 ) => {
   const fetch = fetchCached;
-  const plan = (await fetch(planUrl).then(r => r.json())) as {
-    name: string;
-    permit: string;
-    script: string;
-    bundles: Array<{
-      bundleID: string;
-      entrypoint: string;
-      fileName: string;
-    }>;
-  };
-  assert.equal(plan.name, config.name);
-  const script = await fetch(`${artifacts}/${plan.script}`).then(r => r.text());
-  const permit = await fetch(`${artifacts}/${plan.permit}`).then(r => r.text());
-  const bundles: EndoZipBase64Bundle[] = await Promise.all(
-    plan.bundles.map(b =>
-      fetch(`${artifacts}/${b.bundleID}.json`).then(r => r.json()),
-    ),
-  );
-  return { bundles, evals: [{ js_code: script, json_permits: permit }] };
+
+  try {
+    const planResponse = await fetch(planUrl);
+    if (planResponse.ok) {
+      const plan = (await planResponse.json()) as {
+        name: string;
+        permit: string;
+        script: string;
+        bundles: Array<{
+          bundleID: string;
+          entrypoint: string;
+          fileName: string;
+        }>;
+      };
+
+      assert.equal(plan.name, config.name);
+      const script = await fetch(`${artifacts}/${plan.script}`).then(r =>
+        r.text(),
+      );
+      const permit = await fetch(`${artifacts}/${plan.permit}`).then(r =>
+        r.text(),
+      );
+      const bundles: EndoZipBase64Bundle[] = await Promise.all(
+        plan.bundles.map(b =>
+          fetch(`${artifacts}/${b.bundleID}.json`).then(r => r.json()),
+        ),
+      );
+
+      return { bundles, evals: [{ js_code: script, json_permits: permit }] };
+    }
+  } catch (error) {
+    console.warn(
+      `Plan file not found at ${planUrl}. Falling back to direct artifact detection.`,
```

Comment:
why bother with plan fetching at all I wonder. we could use some more regularity around this stuff but out of scope.

Reply:
Good point